frontend/src/bundles/auth/store/slice.ts

```diff
@@ -3,23 +3,34 @@ import { createSlice } from '@reduxjs/toolkit';
 import { DataStatus } from '~/bundles/common/enums/enums.js';
 import { type ValueOf } from '~/bundles/common/types/types.js';
 import { type UserGetCurrentResponseDto } from '~/bundles/users/users.js';
+import { storage, StorageKey } from '~/framework/storage/storage.js';
 
 import { loadCurrentUser, logout, signIn, signUp } from './actions.js';
 
 type State = {
     dataStatus: ValueOf<typeof DataStatus>;
     user: UserGetCurrentResponseDto | null;
+    isSidebarCollapsed: boolean;
 };
 
 const initialState: State = {
     dataStatus: DataStatus.IDLE,
     user: null,
+    isSidebarCollapsed: false,
 };
 
 const { reducer, actions, name } = createSlice({
     initialState,
     name: 'auth',
-    reducers: {},
+    reducers: {
+        toggleSidebar: (state, action: { payload: boolean }) => {
+            state.isSidebarCollapsed = action.payload;
+            void storage.set(
+                StorageKey.IS_SIDEBAR_COLLAPSED,
+                JSON.stringify(action.payload),
+            );
```

Comment:
its generally a bad practice to handle those things directly in reducer
Maybe we should do it after the action is performed